js/client.js

```diff
@@ -1839,10 +1839,17 @@
 				}
 			}
 			this.$tabbar.html(buf);
-			var $lastLi = this.$tabbar.children().last().children().last();
+			var $lastUl = this.$tabbar.children().last();
+			var $lastLi = $lastUl.children().last();
 			var offset = $lastLi.offset();
 			var width = $lastLi.outerWidth();
-			if (offset.top >= 37 || offset.left + width > $(window).width() - 165) {
+			// 166 here is the difference between the .maintabbar's right margin and the a.button's right margin.
+			if (app.curSideRoom && offset.left + width > $(window).width() - 166) {
+				var margin = $(window).width() + app.curSideRoom.leftWidth - offset.left - width - 310;
+				$lastUl.css("margin-left", margin + "px");
+				offset = $lastLi.offset();
+			}
```

Comment:
If the tabs fit (left + width + 166 < window) then I shouldn't have to change the margin.